plugins/janus_videoroom.c

```diff
@@ -5050,8 +5050,8 @@ gboolean wait_for_pipeline_close(forward_media_type media_type,
                                         end_time)) {
                         	// timeout has passed.
 			g_mutex_unlock (&room->gst_thread_parameters[media_type].gstr.mutex);
-                        JANUS_LOG(LOG_ERR, "wait for pipeline, closed by timeout pipeline  leaved \n");
-                        break;
+                        JANUS_LOG(LOG_ERR, "wait for pipeline, closed by timeout, pipeline-leaved \n");
+                        return FALSE;
```

Comment:
What's the difference it makes?
Anyway it terminates the while loop (either break or return statements),  so this is not the reason for duplicate mutex unlocking.

Reply:
Also, no one uses the return value of this function. Please fix it as well.